enacts/enactstozarr.py

```diff
@@ -41,55 +120,148 @@ def regridding(data, resolution):
+    """
     data = xr.open_mfdataset(
-        netcdf,
-        preprocess = partial(set_up_dims, time_res=time_res),
-        parallel=False
+        paths,
+        preprocess=partial(set_up_dims, time_res=time_res),
+        parallel=False,
     )[var_name]
     if zarr_resolution != None:
-        print("attempting regrid")    
+        print("attempting regrid")
         data = regridding(data, zarr_resolution)
+    return xr.Dataset().merge(data.chunk(chunks=chunks))
 
-    data = data.chunk(chunks=CONFIG['datasets'][time_res]['chunks'])
-    
-    if output_path == None:
-        zarr = f"{CONFIG['datasets'][time_res]['zarr_path']}{input_path}"
+
+def convert(
+    input_path,
+    output_path,
+    var_name,
+    time_res="daily",
+    zarr_resolution=None,
+    chunks={}
+):
+    """Converts a set of ENACTS files into zarr store.
+
+    Either create a new one or append an existing one
+
+    Parameters
+    ----------
+    input_path : str
+        path where the ENACTS nc files are
+    output_path : str
+        path where the zarr store is (to append to) or will be (to create).
+        To create, (last element of the) path is expected not to exist.
+        To append, path is expected to point to a zarr store.
+    var_name : str
+        name of the ENACTS variable in the nc files
+    time_res : str, optional
+        indicates the time resolution of the set of files.
+        Default is "daily" and other option is "dekadal"
+    zarr_resolution : real, optional
+        spatial resolution to regrid to.
+    chunks : int, tuple of int, "auto" or mapping of hashable to int, optional
+        Chunk sizes along each dimension X, Y and T.
+        
+    Returns
+    -------
+    output_path : where the zarr store has been written
+
+    See Also
+    --------
+    calc.read_zarr_data, filename2datetime, nc2xr, xarray.Dataset.to_zarr
+    """
+    print(f"converting files for: {time_res} {var_name}")
+    netcdf = list(sorted(Path(input_path).glob("*.nc")))
+    if Path(output_path).is_dir() :
+        current_zarr = calc.read_zarr_data(output_path)
+        last_T_zarr = current_zarr["T"][-1]
+        last_T_nc = filename2datetime(netcdf[-1], time_res=time_res)
+        if last_T_nc.strftime("%Y%m%d") < last_T_zarr.dt.strftime("%Y%m%d").values :
+            print((
+                f'nc set ({last_T_nc.strftime("%Y%m%d")}) ends before '
+                f'zarrs ({last_T_zarr.dt.strftime("%Y%m%d").values})'
+            ))
+            print("Not changing existing zarr")
+        elif (
+            last_T_nc.strftime("%Y%m%d") ==
+            last_T_zarr.dt.strftime("%Y%m%d").values
+        ) :
+            print(f'both sets end same date: {last_T_nc}')
+            print("Not changing existing zarr")
+        else :
+            print((
+                f'appending nc to zarr from '
+                f'{(last_T_zarr+np.timedelta64(1, "D")).dt.strftime("%Y%m%d").values}'
+                f' to {last_T_nc.strftime("%Y%m%d")}'
```

Comment:
Even when creating a string to be printed, I'm not sure this strftime is helpful. You can just print a datetime, and the default string representation is more readable than %Y%d%d with no delimiters.